File: apps/api/momento_api/models/trash.py
from typing import Optional

from pydantic import BaseModel, ConfigDict, field_serializer
from pydantic.alias_generators import to_camel


class TrashMediaResponse(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
    )

    id: int
    filename: str
    original_filename: str
    media_type: str
    mime_type: Optional[str]
    width: Optional[int]
    height: Optional[int]
    file_size: Optional[int]
    duration_seconds: Optional[float]
    date_taken: Optional[str]
    deleted_at: str
    created_at: str


class TrashListResponse(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
    )

    items: list[TrashMediaResponse]
    total_count: int


class TrashRestoreRequest(BaseModel):
    media_ids: list[int]


class TrashDeleteRequest(BaseModel):
    media_ids: list[int]


class TrashResponse(BaseModel):
    message: str
    affected_count: int

File: apps/api/src/models/trash.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TrashMediaResponse {
    pub id: i64,
    pub filename: String,
    pub original_filename: String,
    pub media_type: String,
    pub mime_type: Option<String>,
    pub width: Option<i32>,
    pub height: Option<i32>,
    pub file_size: Option<i64>,
    pub duration_seconds: Option<f64>,
    pub date_taken: Option<String>,
    pub deleted_at: String,
    pub created_at: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TrashListResponse {
    pub items: Vec<TrashMediaResponse>,
    pub total_count: i64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TrashRestoreRequest {
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TrashDeleteRequest {
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TrashResponse {
    pub message: String,
    pub affected_count: i64,
}
